src/util/error.cpp

```diff
@@ -32,7 +32,9 @@ bilingual_str TransactionErrorString(const TransactionError err)
         case TransactionError::SIGHASH_MISMATCH:
             return Untranslated("Specified sighash value does not match value stored in PSBT");
         case TransactionError::MAX_FEE_EXCEEDED:
-            return Untranslated("Fee exceeds maximum configured by user (e.g. -maxtxfee, maxfeerate)");
+            return Untranslated("Fee exceeds maximum configured by user.");
```

Comment:
Better to keep the option name in the error message:

```suggestion
            return Untranslated("Fee exceeds maximum configured by user (maxtxfee)");
```

Reply:
Yes but I think should come in a follow-up after cleaning up your suggestion here
https://github.com/bitcoin/bitcoin/pull/29278#discussion_r1472790897

I have a local branch that does this on top of this.

Reply:
The reason is that the maximum fee rate check for [`sendrawtransaction`](https://github.com/bitcoin/bitcoin/blob/a11585692e72cac468fb1496ea2c30e4c07f73e5/src/rpc/mempool.cpp#L84-L94) is performed in `BroadcastTransaction` which returned [`MAX_FEE_EXCEEDED`](https://github.com/bitcoin/bitcoin/blob/a11585692e72cac468fb1496ea2c30e4c07f73e5/src/node/transaction.cpp#L71-L79) error type to `sendrawtransaction`.
If I change the `MAX_FEE_EXCEEDED` error message to "Fee exceeds maximum configured by user (maxtxfee)" without changing how it's checked, this would be misleading for `sendrawtransaction` callers and I wanted this to be a minimal wallet change that adds the `maxfeerate` option to the wallet.

However I agree with your comment, I have now added a commit that also checks the fee rate limit in `BroadcastTransaction`,  and changed how `sendrawtransaction` is implemented, and updated the `MAX_FEE_EXCEEDED` transaction error type message.